runtime/lua/vim/diagnostic.lua

```diff
@@ -228,6 +229,16 @@ end
 --- See |nvim_buf_set_extmark()|.
 --- @field virt_text_hide? boolean
 
+--- @class vim.diagnostic.Opts.VirtualLines
+---
+--- Only show diagnostics for the current line.
+--- (default: `false`)
+--- @field current_line? boolean
+---
+--- A function that takes a diagnostic as input and returns a string.
+--- The return value is the text used to display the diagnostic.
+--- @field format? fun(diagnostic:vim.Diagnostic): string
```

Comment:
Doesn't need to be done in this PR, but I think we should change the semantics of the `format` function to allow it to return a `nil` value, where `nil` means the diagnostic is not displayed by that handler.

This could be used to support the "best fit" approach that @mfussenegger mentioned (e.g. by using a `format` function for both virtual text and virtual lines that chooses when and how a given diagnostic should be displayed).

Reply:
Sounds good. I'll make sure to do this in a follow-up PR.

Reply:
> I think we should change the semantics of the format function to allow it to return a nil value, where nil means the diagnostic is not displayed by that handler.

Nice idea, this could also be used for situations like "I want to hide virtual_lines in specific buffers or filetypes".